src/app/utilities/AnnotateUtilities.ts

```diff
@@ -2,17 +2,70 @@
 import jp from 'jsonpath';
 
 /* Import Utilities */
-import { ExtractFromSchema, ExtractClassesAndTermsFromSchema } from 'app/utilities/SchemaUtilities';
+import { ExtractFromSchema, ExtractClassesAndTermsFromSchema, MakeJsonPathReadableString } from 'app/utilities/SchemaUtilities';
 
 /* Import Types */
 import { DigitalSpecimen } from "app/types/DigitalSpecimen";
 import { DigitalMedia } from "app/types/DigitalMedia";
-import { AnnotationFormProperty, Dict } from "app/Types";
+import { AnnotationFormProperty, AnnotationTarget, ParentClass, Dict } from "app/Types";
 
 
 /* Utilities associated with annotating */
 
 
+/**
+ * Function to extract parent classes from a given super class and provide all their associated information
+ * @param params An object containing: the annotationTarget, superClass and parentClasses properties
+ * @returns The extracted parent classes
+ */
+const ExtractParentClasses = (params: {
+    annotationTarget: AnnotationTarget,
+    superClass: DigitalSpecimen | DigitalMedia | Dict
+}) => {
+    const { annotationTarget, superClass } = params;
+
+    /* Base variables */
+    const pathArray: string[] = ['$'];
+    const parents = jp.parse(annotationTarget.jsonPath).slice(1, -1);
+    const parentClasses: ParentClass[] = [];
+
+    /* For each parant, find and extract its class information */
+    parents.forEach((parent, index) => {
+        /* Filter path from schemas */
+        let parentName = parent.expression.value.replace('ods:', '').replace('dwc:', '').replace('dcterms:', '');
+
+        /* Add to pathArray */
+        pathArray.push(parent.expression.value);
+
+        if (parentName.includes('has') || parentName[0] === parentName[0].toUpperCase()) {
+            /* If a parent class is empty, it needs to be targeted, if it is not completely empty, check for indexes */
+            const parentNodes = jp.nodes(superClass, jp.stringify(pathArray).replaceAll('][', ']..[').replaceAll('"', "'"));
+
+            if (!parentNodes.length) {
+                parentClasses.push({
+                    jsonPath: jp.stringify(pathArray).replaceAll('"', "'"),
+                    name: MakeJsonPathReadableString(jp.stringify(pathArray)),
+                    ...(index > 0 && { parentName: pathArray[index] }),
+                    present: false
+                });
+            }
+
+            parentNodes.forEach((parentNode, parentIndex) => {
+                parentClasses.push({
+                    jsonPath: jp.stringify(parentNode.path).replaceAll('"', "'"),
+                    name: MakeJsonPathReadableString(jp.stringify(parentNode.path)),
+                    ...(index > 0 && { parentName: MakeJsonPathReadableString(pathArray[index]) }),
+                    present: (Array.isArray(parentNode.value) && !!parentNode.value.length) || (!Array.isArray(parentNode.value) && typeof (parentNode.value) === 'object'),
```

Comment:
`typeof (parentNode.value) === 'object'` ... javascript doesn't have built-in values for this?

Reply:
How do you mean? The object type is a built in type of JS, here I am checking if it is an object or not. Arrays also count as object types in JS

Reply:
sorry I meant it looks strange to hardcode the string "object" to check if parentNode's value is an object. seems like there should be a better way than to use string comparison

Reply:
you can resolve this if there isn't a better way